packages/lib/passport-interface/test/SubscriptionManager.spec.ts

```diff
@@ -137,6 +199,146 @@ describe("Subscription Manager", async function () {
     }
   });
 
+  it("cloning should work", async function () {
+    const manager = new FeedSubscriptionManager(mockFeedApi);
+
+    const providerUrl = "test url";
+    manager.addProvider(providerUrl, PROVIDER_NAME);
+
+    const feed: Feed = {
+      description: "description",
+      id: "1",
+      name: "test feed",
+      permissions: [],
+      inputPCDType: undefined,
+      partialArgs: undefined,
+      credentialRequest: {
+        signatureType: "sempahore-signature-pcd"
+      }
+    };
+
+    const sub = await manager.subscribe(providerUrl, feed, undefined);
+
+    const action = {
+      type: PCDActionType.ReplaceInFolder,
+      folder: "TEST",
+      pcds: [
+        await SemaphoreIdentityPCDPackage.serialize(
+          await SemaphoreIdentityPCDPackage.prove({
+            identity: new Identity()
+          })
+        )
+      ]
+    };
+
+    manager.setError("1", {
+      type: SubscriptionErrorType.PermissionError,
+      actions: [action]
+    });
+
+    const cleanupListener = manager.updatedEmitter.listen(() => {
+      assert.fail("Listener should never be called.");
+    });
+
+    expect(manager.getProviders()).to.have.length(1);
+    expect(manager.getActiveSubscriptions()).to.have.length(1);
+    expect(manager.getSubscription(sub.id)).to.not.be.undefined;
+
+    const clone = manager.clone();
+
+    expect(clone.getProviders()).to.deep.eq(manager.getProviders());
+    expect(clone.getSubscriptionsByProvider()).to.deep.eq(
+      manager.getSubscriptionsByProvider()
+    );
+    expectSameSubs(manager, clone);
+    expect(clone.getAllErrors()).to.deep.eq(manager.getAllErrors());
+
+    clone.unsubscribe(sub.id); // Would trigger listener if it were on the clone.
+    cleanupListener();
+  });
+
+  it("merging unique data should work", async function () {
```

Comment:
not quite sure what you mean by `unique` here on first read - i see that some providers are not unique, but all provider/feed pairs are unique.